### DISCUSSION TITLE
Managing Tasks while in Non-Admin Account

### DISCUSSION THREAD

**Discussion by JamesIgoe** [dahall/TaskScheduler#984] [Q&A]
As part of a desktop application, I created code that creates, modifies, and deletes scheduled tasks, but it only works in the admin instance, not when under a user. I've tried impersonation that did not solve the problem and writing to/from the Tasks subfolder (C:\Windows\Systems32\Tasks) has been buggy at best.

Any ideas?

**Comment by kedar-hatkar** [comment]
I am using below example to update task scheduler time. Provide valid user credentials having admin rights

using (TaskService taskservice = new TaskService("MachineName", "UserName", "DomainName", "Password"))
                    {
                        var TaskName = "abcd";
                        var task = taskservice.GetTask(TaskName);
                        if (task != null)
                        {
                            var td = task.Definition;
                            td.Triggers[0].Repetition.Interval = TimeSpan.FromMinutes(45);
                            taskservice.RootFolder.RegisterTaskDefinition(TaskName, td, TaskCreation.Update, UserName);
                           ///log Task Updated successfully 
                        }
                        else
                        {
                            /// task doesn't exist in task scheduler"
                        }
                    }
